orcid-core/src/main/java/org/orcid/pojo/ajaxForm/PojoUtil.java

```diff
@@ -141,7 +146,99 @@ public static String createDateSortString(org.orcid.jaxb.model.v3.dev1.common.Fu
+                dateSortString = "X-";
+                dateSortString += endDate.getYear() == null ? "NaN" : endDate.getYear().getValue();
+                if (!PojoUtil.isEmpty(endDate.getMonth()))
+                    dateSortString += "-" + endDate.getMonth().getValue();
+                if (!PojoUtil.isEmpty(endDate.getDay()))
+                    dateSortString += "-" + endDate.getDay().getValue();
+                if(startDate != null) {
+                    dateSortString += startDate.getYear() == null ? "-NaN" : "-" + startDate.getYear().getValue();
+                    if (!PojoUtil.isEmpty(startDate.getMonth()))
+                        dateSortString += "-" + startDate.getMonth().getValue();
+                    if (!PojoUtil.isEmpty(startDate.getDay()))
+                        dateSortString += "-" + startDate.getDay().getValue();
+                }
+            }
+        }
+        
+        return dateSortString;
+    }
+    
+    //Date sort string for V3.dev1 API
+    public static String createDateSortString(org.orcid.jaxb.model.v3.dev1.record.summary.EducationSummary education) {
+        return createDateSortStringForAffiliations(education.getStartDate(), education.getEndDate(), education.getCreatedDate());
+    }
+    
+    public static String createDateSortString(org.orcid.jaxb.model.v3.dev1.record.summary.EmploymentSummary employment) {
+        return createDateSortStringForAffiliations(employment.getStartDate(), employment.getEndDate(), employment.getCreatedDate());
+    }
+    
+    public static String createDateSortString(org.orcid.jaxb.model.v3.dev1.record.Affiliation affiliation) {
+        return createDateSortStringForAffiliations(affiliation.getStartDate(), affiliation.getEndDate(), affiliation.getCreatedDate());
+    }
+    
+    private static String createDateSortStringForAffiliations(org.orcid.jaxb.model.v3.dev1.common.FuzzyDate startDate, org.orcid.jaxb.model.v3.dev1.common.FuzzyDate endDate, org.orcid.jaxb.model.v3.dev1.common.CreatedDate createdDate) {
+        String dateSortString = "";
+        if(startDate == null && endDate == null) {
+            XMLGregorianCalendar date = createdDate.getValue();
+            dateSortString = "Z-" + date.getYear() + "-" + date.getMonth() + "-"  + date.getDay();
+        } else {
+            if(endDate == null) {
+                dateSortString = "Y-";
+                dateSortString += startDate.getYear() == null ? "NaN" : startDate.getYear().getValue();
+                if (!PojoUtil.isEmpty(startDate.getMonth()))
+                    dateSortString += "-" + startDate.getMonth().getValue();
+                if (!PojoUtil.isEmpty(startDate.getDay()))
+                    dateSortString += "-" + startDate.getDay().getValue();
+            } else {
+                dateSortString = "X-";
+                dateSortString += endDate.getYear() == null ? "NaN" : endDate.getYear().getValue();
+                if (!PojoUtil.isEmpty(endDate.getMonth()))
```

Comment:
We should use 00 is there is no value for month or day, so that the sort string is always the same length.

Reply:
@wjrsimpson Fixed